spectractor/simulation/simulator.py

```diff
@@ -133,29 +134,32 @@ def simulate(self, A1=1.0, A2=0., ozone=300, pwv=5, aerosols=0.05, reso=0.,
             The spectrum uncertainties interpolated function in Target units.
 
         """
-        # distance = np.array(self.chromatic_psf.get_distance_along_dispersion_axis(shift_x=shift_x, shift_y=0))
-        # pixels = np.arange(0, parameters.CCD_IMSIZE) - self.x0[0] - shift
-        # self.disperser.D = float(self.header["D2CCD"])
-        # pixels = self.disperser.grating_lambda_to_pixel(lambdas, x0=self.x0, order=1)
+        # find lambdas including ADR effect
         new_x0 = [self.x0[0] - shift_x, self.x0[1]]
         self.disperser.D = D
-        lambdas = self.disperser.grating_pixel_to_lambda(self.pixels - shift_x, x0=new_x0, order=1)
+        distance = self.chromatic_psf.table['Dx_rot']
+        lambdas = self.disperser.grating_pixel_to_lambda(distance - shift_x, x0=new_x0, order=1)
+        distance -= adr_calib(lambdas, self.adr_params, parameters.OBS_LATITUDE)
```

Comment:
+ sign here also ?

Reply:
I suppose.. it would be better to control this function also